src/test/java/core/basesyntax/service/ReportWriterServiceTest.java

```diff
@@ -0,0 +1,85 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import core.basesyntax.service.impl.ReportWriterServiceImpl;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class ReportWriterServiceTest {
+    private static final String READER_FAILURE_MESSAGE = "Cannot read from file {%s}!";
+    private static final String REPORT_FILE_NAME = "src/main/resources/report.csv";
+    private static final String EMPTY_STRING = "";
+    private static final int READ_END_INDEX = -1;
+    private static ReportWriterService reportWriterService;
+
+    @BeforeAll
+    static void setUp() {
+        reportWriterService = new ReportWriterServiceImpl();
+    }
+
+    @Test
+    void write_correctInputData_Ok() {
+        String report = """
+                fruit,quantity
+                banana,152
+                apple,90
+                """;
+        reportWriterService.writeReport(report, REPORT_FILE_NAME);
+        String actual = readReportFile();
+        assertEquals(report, actual);
+        boolean isDeleted = new File(REPORT_FILE_NAME).delete();
+        assertTrue(isDeleted);

```

Comment:
1) We should not **assert** if we deleted this file, we do not test File.delete() method here, we test our components
2) Usually, such logic is extracted in afterEach/afterAll method